sklearn/utils/validation.py

```diff
@@ -129,28 +129,31 @@ def _is_arraylike(x):
 
 def _num_samples(x):
     """Return number of samples in array-like x."""
+    message =  'Expected sequence or array-like, got %s' % type(x)
     if hasattr(x, 'fit') and callable(x.fit):
         # Don't get num_samples from an ensembles length!
-        raise TypeError('Expected sequence or array-like, got '
-                        'estimator %s' % x)
+        raise TypeError(message)
+
```

Comment:
Some tests of functions relying on `_num_samples` internally check that 'estimator' is in the error message.
We can either change the expected message in those tests or keep the special case when x is an estimator.

Reply:
And since that they are public utils, we should not break the backward compatibility.

Reply:
I changed the test. @glemaitre are you saying changing the text of an error message is a backward-incompatible change? I don't think we need to be that strict and I don't think we have been that strict in the past.
I have no idea how I would deprecate the content of an error message.

Reply:
I see that this is in `validation.py` so it is not an issue if the failure is not raised within `check_estimator`

My reasoning was linked to something that we did in the past: https://github.com/scikit-learn/scikit-learn/pull/13013#discussion_r252445399

Reply:
I only had to change ``test_validation``, not the estimator checks. That means it's not used in ``check_estimator``, right? Otherwise those tests would be failing.

Reply:
True

Reply:
merge?